neuro_flow/context.py

```diff
@@ -1223,6 +1223,15 @@ async def create(  # type:ignore[override]
 
         return cast(_CtxT, replace(ctx, _prefix=prefix, _prep_tasks=prep_tasks))
 
+    async def get_output_needs(self) -> AbstractSet[FullID]:
+        assert isinstance(self._ast, ast.BatchAction)
+        if self._ast.outputs and self._ast.outputs.needs is not None:
+            return {
+                self._prefix + (await need.eval(self),)
+                for need in self._ast.outputs.needs
+            }
+        return self.graph.keys()
```

Comment:
I doubt if we need this branch.
Enforcing the `needs` attribute for outputs looks like a sane idea.
What do you think?

Reply:
Both options are OK for me, I have no preference here. On other same, "not enforced -> enforced" is breaking change, so it's better to make it enforced now, and change this if devops tell us to do so.

Reply:
Agree. 
Please select the stricter option, we can relax it later on the demand.

Reply:
Done

Reply:
I believe the function should return an empty list if no outputs were specified in action's YAML

Reply:
OK, but this will make all actions without outputs never fail.